src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -554,6 +555,18 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
 
                             continue;
 
+                        case "sourcelink":
+                            value = RemoveQuotesAndSlashes(value);
+                            if (string.IsNullOrEmpty(value))
+                            {
+                                AddDiagnostic(diagnostics, ErrorCode.ERR_NoFileSpec, arg);
+                            }
+                            else
+                            {
+                                sourceLink = ParseGenericPathToFile(value, diagnostics, baseDirectory);
```

Comment:
I don't think so. The content is opaque to the compiler similarly to custom attributes or resources. It just encodes it as is and the tool that uses the information (the debugger) will do the validation.